jscomp/ml/typedecl.ml

```diff
@@ -462,7 +455,13 @@ let transl_declaration ~typeRecordAsObject env sdecl id =
                 | {ld_name = {txt = "..."}; ld_type} :: rest, _ :: rest' ->
                   (match Ctype.extract_concrete_typedecl env (extract ld_type.ctyp_type) with
                     (_p0, _p, {type_kind=Type_record (fields, _repr)}) ->
-                      process_lbls (fst acc @ (fields |> List.map mkLbl), snd acc @ fields) rest rest'
+                      process_lbls (fst acc @ (fields |> List.map(fun (lbl: Types.label_declaration) ->
```

Comment:
How about reverting this block of changes too.
I have already checked that it works on the test examples, so I'm not sure why there are changes here.
Or if they're needed, then there's no test to exercise them.
So maybe they're needed, but if they are, let's test them.

Reply:
Right, I get it now. I thought _this_ was the issue, but it's just that the list of names of optional labels is calculated before the spreads. I get it now, fixing.